tests/python/pants_test/engine/legacy/test_graph.py

```diff
@@ -78,7 +78,7 @@ def test_invalidate_fsnode_incremental(self):
 
       # Invalidate the '3rdparty/python' DirectoryListing, the `3rdparty` DirectoryListing,
       # and then the root DirectoryListing by "touching" files/dirs.
-      for filename in ('3rdparty/python/BUILD', '3rdparty/python', 'non_existing_file'):
+      for filename in ('3rdparty/python/BUILD', '3rdparty/python', ''):
```

Comment:
But thinking more about it, that special case should probably be implemented by the `fs_service` (ie, causing it to spit out the root in addition to files at the root).